internal/tiltfile/tiltfile.go

```diff
@@ -74,15 +75,66 @@ func Load(ctx context.Context, filename string, matching map[string]bool, logs i
 		return nil, model.Manifest{}, nil, err
 	}
 
-	yamlManifest := model.Manifest{}
+	var remainingYamlManifest model.Manifest
 	if len(unresourced) > 0 {
-		yamlManifest, err = k8s.NewK8sOnlyManifest(unresourcedName, unresourced)
+		var podCreators []model.Manifest
+		podCreators, remainingYamlManifest, err = k8sManifestsForPodCreators(ctx, unresourced)
 		if err != nil {
 			return nil, model.Manifest{}, nil, err
 		}
+		manifests = append(manifests, podCreators...)
+	}
+
+	// TODO(maia): `remainingYamlManifest` should be processed just like any
+	// other manifest (i.e. get rid of "global yaml" concept)
+	return manifests, remainingYamlManifest, s.configFiles, err
+}
+
+// k8sManifestsForPodCreators groups the given k8s entities into pod creator clusters
+// (e.g. DeploymentA is a pod creator, and ServiceA would belong to its pod creator
+// cluster b/c it applies to pods created by DeploymentA); returns a k8s-only manifest
+// per pod creator cluster, and a single manifest containing the rest of the entities
+// (will be deployed/displayed in HUD as a unit).
+func k8sManifestsForPodCreators(ctx context.Context, entities []k8s.K8sEntity) (podCreators []model.Manifest,
+	remaining model.Manifest, err error) {
```

Comment:
i'm not sure that you're using the return parameter names here? i don't think we should do this just for documentation purposes, because it changes the semantics of your program and makes errors harder to catch, like in the mistake below.